### DISCUSSION TITLE
Get name and result of each test

### DISCUSSION THREAD

**Discussion by dudziakm** [cypress-io/cypress#15047] [Questions and Help] [answered]
I am looking for an option to get the name and the result (fail/pass) of each test to save this to an external file.

How can I get this information easily?

Currently, I can get only the title within the `it()` section with:

`cy.log(this.test.title)`

however, I am not able to get the result there.

It would be ideal to have this option from `afterEach()` section. So that in one place I can save the title and result.

Thanks!

**Comment by bahmutov** [comment]
Take a look at https://glebbahmutov.com/blog/cypress-tips-and-tricks/#use-test-names-when-creating-data and https://glebbahmutov.com/blog/cypress-tips-and-tricks/#get-test-status You will find all this information from the `this` object in the `afterEach` hook

```js
afterEach(function () {
  console.log(this)
  debugger
})
```

**Reply by dudziakm** [reply]
What I am able to see in the console is:
![image](https://user-images.githubusercontent.com/10773170/114154409-aa4b5900-9920-11eb-825e-c20cdf0dfb2f.png)

There is no information about test or currentTest :(

**Reply by JSoet** [reply]
Note that you need to use the `function () {` method of writing a function in order to have `this` available, I changed my function into an arrow function, since that was the standard in the file I was editing, but [arrow functions don't have access to `this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions). For me the this.currentTest object then showed up

**Comment by sbhattathiri** [comment]
```
afterEach(function () {
      console.log(this.currentTest.title)
      console.log(this.currentTest.state)
      console.log(__filename)
  })
```

**Comment by emilyrohrbough** [comment, marked as answer]
Take a look at the [After Spec Run API](https://docs.cypress.io/api/plugins/after-spec-api).

**Comment by csvan** [comment]
Is there really no way to access the current test context when using arrow functions?
